src/utilities/simIncludeGravBody.py

```diff
@@ -15,399 +14,496 @@
+        mu=4.902799e12,
+        radEquator=1738100.0,
+        spicePlanetFrame="IAU_moon",
+    ),
+    "mars": BodyData(
+        identifier="mars",
+        planetName="mars_planet_data",
+        displayName="mars",
+        modelDictionaryKey="",
+        mu=4.28283100e13,
+        radEquator=3396190.0,
+        spicePlanetFrame="IAU_mars",
+    ),
+    "mars barycenter": BodyData(
+        identifier="mars barycenter",
+        planetName="mars barycenter_planet_data",
+        displayName="mars barycenter",
+        modelDictionaryKey="",
+        mu=4.28283100e13,
+        radEquator=3396190.0,
+        spicePlanetFrame="IAU_mars",
+    ),
+    "jupiter barycenter": BodyData(
+        identifier="jupiter barycenter",
+        planetName="jupiter barycenter_planet_data",
+        displayName="jupiter",
+        modelDictionaryKey="",
+        mu=1.266865349093058e17,
+        radEquator=71492000.0,
+        spicePlanetFrame="IAU_jupiter",
+    ),
+    "saturn": BodyData(
+        identifier="saturn",
+        planetName="saturn barycenter_planet_data",
+        displayName="saturn",
+        modelDictionaryKey="",
+        mu=3.79395000e16,
+        radEquator=60268000.0,
+        spicePlanetFrame="IAU_saturn",
+    ),
+    "uranus": BodyData(
+        identifier="uranus",
+        planetName="uranus barycenter_planet_data",
+        displayName="uranus",
+        modelDictionaryKey="",
+        mu=5.79396566e15,
+        radEquator=25559000.0,
+        spicePlanetFrame="IAU_uranus",
+    ),
+    "neptune": BodyData(
+        identifier="neptune",
+        planetName="neptune barycenter_planet_data",
+        displayName="neptune",
+        modelDictionaryKey="",
+        mu=6.83509920e15,
+        radEquator=24764000.0,
+        spicePlanetFrame="IAU_neptune",
+    ),
+}
+
+
+@runtime_checkable
+class WithGravField(Protocol):
+    gravField: Any  # cannot be GravityEffector because SWIG classes dont give type info
+
+
+class gravBodyFactory:
     """Factory cass to create gravitational bodies."""
-    def __init__(self, bodyNames=None):
-        self.spicePlanetNames = []
-        self.spicePlanetFrames = []
-        self.gravBodies = OrderedDict()
-        self.spiceObject = None
-        self.spiceKernelFileNames = []
-        self.epochMsg = None
+
+    def __init__(self, bodyNames: Iterable[str] = []):
+        self.spicePlanetFrames: List[str] = []
+        self.gravBodies: Dict[str, gravityEffector.GravBodyData] = {}
+        self.spiceObject: Optional[spiceInterface.SpiceInterface] = None
+        self.spiceKernelFileNames: List[str] = []
+        self.epochMsg: Optional[messaging.EpochMsg] = None
         if bodyNames:
             self.createBodies(bodyNames)
 
-    def createBodies(self, bodyNames):
+    def addBodiesTo(
+        self,
+        objectToAddTheBodies: Union[gravityEffector.GravityEffector, WithGravField],
+    ):
+        """Can be called with a GravityEffector or an object that has a gravField
```

Comment:
The gravity bodies are actually always added to a `GravityEffector`. However, for convinience, we also allow passing an object that has a `gravField` variable which points to a `GravityEffector`. This keeps the function generic (so that users may add the bodies to `GravityEffector`s that exist in contexts that are not `Spacecraft` or `SpacecraftSystem`, such as custom `DynamicObject`), but it makes the syntax easier for the most common use-case, which are `Spacecraft` or `SpacecraftSystem`.
```
sc = spacecraft.Spacecraft()
custom = myCustomObject.MyCustomObject()

gravFactory.addBodiesTo(sc) # ok
gravFactory.addBodiesTo(sc.gravField) # ok, same as above, but we are passing the GravityEffector directly
gravFactory.addBodiesTo(custom.myGravityField) # ok, the custom DynamicObject stores its GravityEffector in a different place than the builtin Spacecraft and SpacecraftSystem, so we have to pass the GravityEffector
```